src/net/sourceforge/kolmafia/request/CoinMasterRequest.java

```diff
@@ -260,23 +260,56 @@ public static void parseResponse(
       CoinMasterRequest.parseBalance(data, responseText);
       return;
     }
-    String shopId = NPCPurchaseRequest.getShopId(urlString);
 
     String buy = data.getBuyAction();
     String sell = data.getSellAction();
-    String buyURL = data.getBuyURL();
-    String sellURL = data.getSellURL();
-    if (buy != null
-        && action.equals(buy)
-        && (buyURL == null || shopId == null || buyURL.endsWith(shopId))
-        && !responseText.contains("You don't have enough")
-        && !responseText.contains("Huh?")) {
-      CoinMasterRequest.completePurchase(data, urlString);
-    } else if (sell != null
-        && action.equals(sell)
-        && (sellURL == null || shopId == null || sellURL.endsWith(shopId))
-        && !responseText.contains("You don't have that many")) {
-      CoinMasterRequest.completeSale(data, urlString);
+
+    if (buy == null && sell == null) {
+      // You can neither buy nor sell from this Coinmaster? Coding error!
+      return;
```

Comment:
sounds like a good target for a consistency test.
```java
for (CoinmasterData coinmaster : CoinmasterRegistry.COINMASTERS) {
  assertTrue(coinmaster.getBuyAction != null || coinmaster.getSellAction() != null)
}
```

Reply:
I added a consistency test. Turns out there is exactly one Coinmaster which has neither a buyAction nor a sellAction - AWOLQuartermasterRequest. The URL for buying something is "inv_use.php?whichitem=5116&ajax=1".

It also has buyItems. I made the test check for buyAction, and if it fails, for buyItems.

Perhaps my tests in CoinMasterRequest should be checking for something other than butAction, but considering the single Coinmaster which does NOT have that is way obsolete, I don't feel like trying to figure that out at this time.

I have a feeling that future Coinmasters will all be shop.php.